src/ComboBox/index.jsx

```diff
@@ -108,583 +104,547 @@ function optionsFormatted( filteredOptionsIds, originalOptions )
-                const contHeight = scrollBoxRef.offsetHeight;
+                const contHeight = scrollBoxRef.current.offsetHeight;
 
-                const min = scrollBoxRef.scrollTop;
-                const max = min + ( scrollBoxRef.offsetHeight - elHeight );
+                const min = scrollBoxRef.current.scrollTop;
+                const max = min +
+                  ( scrollBoxRef.current.offsetHeight - elHeight );
 
                 if ( pos < min )
                 {
-                    scrollBoxRef.scrollTop = pos;
+                    scrollBoxRef.current.scrollTop = pos;
                 }
                 else if ( pos > max )
                 {
-                    scrollBoxRef.scrollTop = pos - ( contHeight - elHeight );
+                    scrollBoxRef.current.scrollTop =
+                      pos - ( contHeight - elHeight );
                 }
             }
         }
-    }
+    } );
 
-    focus()
+    const focus = useCallback( () =>
     {
-        this.inputRef.current.focus();
-    }
+        inputRef.current.focus();
+    }, [ inputRef.current ] );
 
-    filterOptions( tags )
+    const handleFocus = useCallback( () =>
     {
-        return this.state.options.filter( option =>
-            !tags.includes( option.text ) );
-    }
-
-    handleFocus()
-    {
-        this.focus();
+        focus();
 
-        if ( this.props.isSearchable )
+        if ( isSearchable )
         {
-            this.setState( { searchValue: '' } );
+            setSearchValue( '' );
         }
-    }
+    }, [ isSearchable ] );
 
-    handleChangeInput( e )
+    const handleChangeInput = useCallback( ( e ) =>
     {
         e.stopPropagation();
-        const { value } = e.target;
 
-        const { onChangeInput } = this.props;
+        const searchValueToUse = ( e.target.value || '' ).toLowerCase();
+
         if ( typeof onChangeInput === 'function' )
         {
-            onChangeInput( { value }, e );
+            onChangeInput( { value: e.target.value }, e );
         }
 
-        this.setState( { searchValue: value } );
-    }
+        setSearchValue( searchValueToUse );
+    }, [ flatOptions, onChangeInput ] );
 
-    handleClickIcon()
+    const handleClickIcon = useCallback( () =>
     {
-        this.focus();
-        this.setState( prevState => ( { isOpen: !prevState.isOpen } ) );
-    }
+        focus();
+        setIsOpen( !isOpen );
+    }, [ isOpen ] );
 
-    handleClick()
+    const handleClick = () =>
     {
-        this.setState( { isOpen: true  } );
-    }
+        setIsOpen( true );
+    };
 
-    handleClickOption( { id: prefixedId } )
+    const handleClickOption = useCallback( ( { id : optId } ) =>
     {
-        this.setState( ( { id, selection } ) =>
-        {
-            const optId = removePrefix( prefixedId, id );
-            const { isMultiselect, onChange } = this.props;
+        const unprefixedId = removePrefix( optId, id );
+
+        let newSelection = !isReadOnly ? getOption(
```

Comment:
Calling `getOption` here is redundant (I think) since we already have the `id` of the new option (`unprefixedId`)